dev/com.ibm.ws.concurrent_fat_quartz/test-applications/quartzapp/src/concurrent/fat/quartz/web/QuartzTestServlet.java

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package concurrent.fat.quartz.web;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import jakarta.annotation.Resource;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+
+import org.junit.Test;
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.SimpleTrigger;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.quartz.impl.matchers.KeyMatcher;
+import org.quartz.impl.matchers.NameMatcher;
+
+import componenttest.app.FATServlet;
+
+@SuppressWarnings("serial")
+@WebServlet("/*")
+public class QuartzTestServlet extends FATServlet {
+
+    // Maximum number of nanoseconds to wait for a task to finish.
+    private static final long TIMEOUT_NS = TimeUnit.MINUTES.toNanos(2);
+
+    /**
+     * Defines a resource reference that Quartz jobs can try to look up to verify that
+     * application component context is available.
+     */
+    @Resource(name = "java:module/env/concurrent/quartzExecutorRef", lookup = "concurrent/quartzExecutor")
+    ExecutorService executorRef;
+
+    /**
+     * Quartz scheduler
+     */
+    private Scheduler scheduler;
+
+    @Override
+    public void destroy() {
+        if (scheduler != null)
+            try {
+                scheduler.shutdown();
```

Comment:
Do you need to additionally set `scheduler=null` here so that if destroy is called again we can skip the `shutdown()` call?

Reply:
sure, I can add that to be safe